parser.y

```diff
@@ -1279,6 +1279,17 @@ AlterTableSpec:
 		}
 		$$ = op
 	}
+|	"CONVERT" "TO" CharsetKw "DEFAULT" OptCollate
```

Comment:
> It used to parse the CONVERT TO CHARACTER SET chaset_name into [DEFAULT] CHARACTER SET syntax. I think we just want to change only the default character set for a table...

> So I reserved it.

I think we need to support all syntaxes rather than `CONVERT TO CHARSET DEFAULT`. Please update the codes here for specified character name.

BTW, the current implementation of `ALTER TABLE ... DEFAULT CHARSET` in TiDB is wrong, I think we need to fix it later.